File: src/main/java/com/todolist/task/service/domain/models/Task.java
package com.todolist.task.service.domain.models;

import com.todolist.task.service.domain.enums.Priority;
import com.todolist.task.service.domain.enums.TaskStatus;
import java.time.Instant;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;

@Getter
@Setter
@Builder
@NoArgsConstructor
@AllArgsConstructor
public class Task {

  private String title;
  private String description;
  private Priority priority;
  private TaskStatus status;
  private Instant dueDate;
  private boolean deleted;
  private Instant createdTime;

  private String createdBy;

  private Instant updatedTime;

  private String updatedBy;
}

File: src/main/kotlin/com/todolist/task/service/domain/models/Task.kt
package com.todolist.task.service.domain.models

import com.todolist.task.service.domain.enums.Priority
import com.todolist.task.service.domain.enums.TaskStatus
import java.time.Instant

data class Task(
    val id: String = "",
    val title: String,
    val description: String,
    val priority: Priority,
    var status: TaskStatus,
    val dueDate: Instant,
    var deleted: Boolean = false,
    val createdTime: Instant,
    val createdBy: String,
    val updatedTime: Instant,
    var updatedBy: String
) {
    fun deleteTask(user: String) {
        this.deleted = true
        this.updatedBy = user
    }

}
